### ISSUE TITLE
Using this module I am unable to authenticate to the gke private cluster

### ISSUE THREAD

**Issue by Rajchirag1993** [terraform-google-modules/terraform-google-kubernetes-engine#1417]
### TL;DR

Hello,

I am using this module to authenticate to gke private cluster but I am facing i/o timeout error.

Can you please guide me what is missing ??/

### Expected behavior

I am trying to deploy kubernetes resources through terraform 

### Observed behavior

module.terraform-google-kubernetes-engine.kubernetes_namespace.my-first-namespace: Creating...
module.terraform-google-kubernetes-engine.kubernetes_namespace.my-first-namespace: Still creating... [10s elapsed]
module.terraform-google-kubernetes-engine.kubernetes_namespace.my-first-namespace: Still creating... [20s elapsed]
module.terraform-google-kubernetes-engine.kubernetes_namespace.my-first-namespace: Still creating... [30s elapsed]
╷
│ Error: Post "http://X.X.X.X:443/api/v1/namespaces": dial tcp X.X.X.X:443: i/o timeout
│
│   with module.terraform-google-kubernetes-engine.kubernetes_namespace.my-first-namespace,
│   on .terraform/modules/terraform-google-kubernetes-engine/manifests.tf line 27, in resource "kubernetes_namespace" "my-first-namespace": 
│   27: resource "kubernetes_namespace" "my-first-namespace" {
│

### Terraform Configuration

```hcl
Basic kubernetes private cluster configuration
```

### Terraform Version

```sh
Terraform v1.0.11
```

**Comment by richiefrich** [comment]
I dont see your configuration for the module but are you using this option?

deploy_using_private_endpoint

**Comment by bharathkkb** [comment]
+1 to @richiefrich, sample config would help us understand the reason. If it is using private endpoint, you will need direct network connectivity to the apiserver IP.

**Comment by Rajchirag1993** [comment]
Hello Stephen,

I am using the same template of your module, getting connection refused and
connection timed out errors.

Yes you are correct I am using private cluster configuration which includes

1) master authorised networks
2) private endpoint
3) private nodes
4) network policy

Please let me know how to authenticate to gke private cluster from
terraform to spin up kubernetes resources like deployment and services.
